### PR TITLE
SignInViewController with Texture

### PR DESCRIPTION
**Pull request by Kharchevskyi**

### CODE REVIEW

**Comment by tomholub** [comment]
issue #58

File: FlowCrypt/Controllers/SignIn/Models/SignInOption.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  OptionsButton.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 22.10.2019.
+//  Copyright © 2019 FlowCrypt Limited. All rights reserved.
+//
+
+import Foundation
+
+enum SignInOption: String, CaseIterable {
```
Comment (tomholub) [inline FlowCrypt/Controllers/SignIn/Models/SignInOption.swift]: Call `SignInLinks`?

File: FlowCrypt/Controllers/SignIn/SignInViewController.swift [hunk 2]
```diff
@@ -4,61 +4,36 @@
...
 import GoogleSignIn
 import UIKit
+import AsyncDisplayKit
 
-final class SignInViewController: UIViewController {
-    // TODO: Inject as a dependency
-    private let userService = UserService.shared
+final class SignInViewController: ASViewController<ASTableNode> {
+    enum Parts: Int, CaseIterable {
+        case options, logo, description, gmail, outlook
```
Comment (tomholub) [inline FlowCrypt/Controllers/SignIn/SignInViewController.swift]: here `options` could be called `links`

File: FlowCrypt/Controllers/SignIn/Nodes/OptionButtonNode.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
+//
```
Comment (tomholub) [inline FlowCrypt/Controllers/SignIn/Nodes/OptionButtonNode.swift]: Could be called `LinkButtonNode`

File: FlowCrypt/Controllers/SignIn/SignInViewController.swift [hunk 4]
```diff
@@ -72,21 +47,63 @@ final class SignInViewController: UIViewController {
...
+        return Parts.allCases.count
+    }
+
+    func tableNode(_ tableNode: ASTableNode, nodeBlockForRowAt indexPath: IndexPath) -> ASCellNodeBlock {
+        let imageHeight = tableNode.bounds.size.height * 0.2
+
+        return { [weak self] in
+            guard let self = self, let part = Parts(rawValue: indexPath.row) else { return ASCellNode() }
+            switch part {
+            case .options:
```
Comment (tomholub) [inline FlowCrypt/Controllers/SignIn/SignInViewController.swift]: links

**Comment by tomholub** [comment]
app works well, layout looks good too

**Comment by Kharchevskyi** [comment]
Fix all comments in latest commit.

**Pull request merged**